### ISSUE TITLE
FCM: iOS: is not displayed notification's message

### ISSUE THREAD

**Issue by newapx** [EddyVerbruggen/nativescript-plugin-firebase#530]
When sending the message for iOS application receives it only if to be in foreground.
If application in background or killed, banner with incoming notification's message is not displayed by and nothing appears in Notifications swipe down list.
After application opening all earlier sent messages (which were not displayed) come to applications.

Such behavior on both of my iPad devices (iOS 9) and iPhone (iOS 10).
All settings are made as it is specified in Readme.
Batteries on devices are charged.
For application the Updating of Content mode is included

I checked sending the message with:
- only with section `notification`
- only with section `data`
- including both sections
- also I checked sending the message specified in the section Testing

In all cases the behavior described above.

**Comment by b4rtt** [comment]
Same problem here.

**Comment by andreacappadona17** [comment]
Same issue here. It was working in production with version 3.x.x, but it stopped working after updating the plugin

**Comment by henrychavez** [comment]
The problem was solved #523 but not release it yet. 

to be specific the problem is the version of Firebase if you install the latest version of Firebase will solve your problem .

**Comment by EddyVerbruggen** [comment]
@henrychavez Actually it is, because that issue mentions it's fixed in Firebase 4.3.0, and the current version of this plugin uses Firebase 4.4.0.

**Comment by henrychavez** [comment]
My mistake.

I notice that the lates version of Firebase is 4.5.0 I hope the problem is solved in that version

**Comment by b4rtt** [comment]
@henrychavez It doesn't help me. I still get notifications only when app is running :(

**Comment by ShawnPavel** [comment]
Just tried out the newest version of firebase (4.2.1) and all seems to be working for me again. Looks like [this pull request](https://github.com/EddyVerbruggen/nativescript-plugin-firebase/pull/522) did the fix.

**Comment by b4rtt** [comment]
@ShawnPavel @EddyVerbruggen @henrychavez Could someone help me please. This is how my repository looks. I still getting notifications only when app is opened. When I close it, or minimalize it it stop to work. Notifications is show after I again open app.

https://github.com/b4rtt/no_working_notifications

Thank you so much. I am kinda frustrated cuz I try to make it works almost month :(

**Comment by henrychavez** [comment]
@b4rtt I already test it with the latest version of the plugin 4.2.1 and works as expected.

Try deleting your node_modules and platforms. Also move the firebase.init() to ngAfterViewInit().

As Shawn said, the problem with background notification was solved in this pull request #522

**Comment by b4rtt** [comment]
@henrychavez I tried it and it's still the same :(

**Comment by henrychavez** [comment]
@b4rtt  yes!
probably your certificates are the problem. Check your firebase console too.

did you remove and reinstall the plugin?

**Comment by bozhidarc** [comment]
I have the exact same problem, too. If the app is opened in foreground everything ok. I receive all the push messages from the FCM console and from my backend, but if its closed I receive nothing. 
I'm trying to run this plugin successfully from weeks by reading and experimenting and nothing happens.
Is there a working version even if it's old. I only need the notification?

My env is:
- nativescript 3.3.1
- tns-core-modules 3.3.0
- tns-ios 3.3.0
- Angular 4.4.1
- nativescript-plugin-firebase 2.4.1

I'm calling in `main.ts` `firebase.init(...)` exactly before `platformNativeScriptDynamic().bootstrapModule(AppModule);`

I'm adding the message receive callback in my `app.component.ts` in the 
```
ngAfterViewInit(): void {
		firebase
			.addOnMessageReceivedCallback((message: any) => {
				console.log(message)
			})
			.catch(error => {
				console.error(error);
			});
	}
```

**Comment by angjnu** [comment]
I am getting following error!

ERROR file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:569:26: Unhandled Promise rejection: Error Domain=com.google.fcm Code=2001 "FIRMessaging is already connected" UserInfo={NSLocalizedFailureReason=FIRMessaging is already connected} ; Zone: <root> ; Task: null ; Value: Error Domain=com.google.fcm Code=2001 "FIRMessaging is already connected" UserInfo={NSLocalizedFailureReason=FIRMessaging is already connected} undefined
CONSOLE ERROR file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:571:22: Error: Uncaught (in promise): Error Domain=com.google.fcm Code=2001 "FIRMessaging is already connected" UserInfo={NSLocalizedFailureReason=FIRMessaging is already connected}

**Comment by bozhidarc** [comment]
After alot of reading and trying different combination of where to place the init function to removing all development certificates this answer helped me most https://discourse.nativescript.org/t/how-to-implement-push-notification-on-nativescript-angular-2/173/23.

The  `firebase.init()` should be wrapped in `setTimeout(..., 3000)` function.
@angjnu , I hade the same error messages, but I think this is not critical because everything works with it.

**Comment by Pant76** [comment]
Same problem....i tried this as @bozhidarc  suggests:
https://discourse.nativescript.org/t/how-to-implement-push-notification-on-nativescript-angular-2/173/23.
But i'm using javascript (no andular no typescript)
No way to get it working!

**Comment by angjnu** [comment]
I have used

 "nativescript-plugin-firebase": "^3.12.0" 

and I was able to receive the notification in background but I made some changes in main.ts  then I received following errors.

CONSOLE ERROR file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:569:26: Unhandled Promise rejection: undefined is not an object (evaluating 'firebase.
CONSOLE ERROR file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:571:22: Error: Uncaught (in promise): TypeError: undefined is not an object (evaluating 'firebase.
CONSOLE LOG file:///app/tns_modules/@angular/core/./bundles/core.umd.js:2957:20: Angular is running in the development mode. Call enableProdMode() to enable the production mode.


my package.json file have following dependencies: 
"dependencies": {
		"@angular/animations": "~4.4.5",
		"@angular/common": "~4.4.5",
		"@angular/compiler": "~4.4.5",
		"@angular/core": "~4.4.5",
		"@angular/forms": "~4.4.5",
		"@angular/http": "~4.4.5",
		"@angular/platform-browser": "~4.4.5",
		"@angular/router": "~4.4.5",
		"nativescript-angular": "~4.4.1",
		"nativescript-plugin-firebase": "^3.12.0",
		"nativescript-theme-core": "~1.0.4",
		"reflect-metadata": "~0.1.8",
		"rxjs": "~5.4.2",
		"tns-core-modules": "~3.1.1",
		"zone.js": "~0.8.2"
	},
	"devDependencies": {
		"nativescript-dev-typescript": "~0.5.0",
		"typescript": "~2.4.2"
	}

**Comment by erkanarslan** [comment]
This is still a problem. I can receive push messages when app is on foreground but I don't receive them until I start the app. When I start the application, all FCM messages sent when app was on background are received by the application. It's like FCM service froze until app started again.

**Comment by EddyVerbruggen** [comment]
Like I ask everyone with this issue: share a repo and I’ll take a look.

**Comment by erkanarslan** [comment]
@EddyVerbruggen I don't have a repository on Github, so I sent the source code files to your gmail address.

It is better if I provide more information. I was able to get display notifications when iOS app was on background. At that time I was sending FCM messages that contain both notification and data. Then, we changed our server code and now we only send data messages. Our FCM messages don't have notification part anymore. We are trying to create display messages with nativescript-local-notifications library. When we get an FCM message from server, we process it and create a local notification. I am able to create that in Android but not successful on iOS.

On iOS, onMessageReceivedCallback is not called until app started. It seems that FCM notifications are stored somewhere while app is in background and are received by onMessageReceivedCallback immediately after app started (brought to foreground). Because of that, we cannot create local notifications when app is in background.
